src/Chainweb/CutDB.hs

```diff
@@ -328,7 +340,7 @@ withCutDb
     -> WebBlockHeaderStore
     -> WebBlockPayloadStore cas
     -> RocksDbCas CutHashes
-    -> (CutDb cas -> IO a)
+    -> (forall cas' . PayloadCasLookup cas' => CutDb cas' -> IO a)
```

Comment:
This is the main change of this PR. Almost everything else is a consequence of this change.